src/profile-logic/js-tracer.js

```diff
@@ -40,7 +67,37 @@ export function getJsTracerTiming(
     tracerEventIndex++
   ) {
     const stringIndex = jsTracer.events[tracerEventIndex];
-    const displayName = stringTable.getString(stringIndex);
+    const column = jsTracer.columns[tracerEventIndex];
+    const line = jsTracer.lines[tracerEventIndex];
+
+    // By default we use the display name from JS tracer, but we may update it if
+    // we can figure out more information about it.
+    let displayName = stringTable.getString(stringIndex);
+
+    // We may have deduced the funcIndex in the keyToFuncIndex Map.
+    let funcIndex: null | IndexIntoFuncTable = null;
+
+    if (column !== null && line !== null) {
+      // There is both a column and line number for this script. Look up to see if this
+      // script location has a function in the sampled data. This is a simple way
+      // to tie together the JS tracer information with the Gecko profiler's stack
+      // walking.
+      const key = `${displayName}:${line}:${column}`;
+      const funcIndexInMap = keyToFuncIndex.get(key);
+
+      if (funcIndexInMap !== undefined) {
+        if (funcIndexInMap === null) {
+          // This is probably a failure case in our source information.
```

Comment:
or maybe there are really multiple functions in one line? :)

Reply:
This would be a match on both the line AND column, which is a bit nonsensical.